### PR TITLE
Adds links to categories on organisation page

### PR DESCRIPTION
**Pull request by bryszard**
PT - https://www.pivotaltracker.com/story/show/78632274.

We've been working on this with @tongkiat from AV102. We've managed to add some cukes for now. I'll be working on the implementation and more tests (maybe with @tongkiat if he wants to join me).

### CODE REVIEW

File: app/views/organisations/_show.html.erb [hunk 1]
```diff
@@ -35,15 +35,27 @@
...
+          </p>
         </li>
         <li id="how_they_help">
           <h5><strong>How they help</strong></h5>
-          <p><%= org.categories.how_they_help.pluck(:name).join(', ') %></p>
+          <p>
+          <% org.categories.how_they_help.collect do |category| %>
+          <%= link_to category.name, organisations_search_path(what_id: '', who_id: '', how_id: category.id, q: '') %>
+          <% end.join(', ') %>
+          </p>
```
Comment (tansaku) [inline app/views/organisations/_show.html.erb]: good show - could pull this set of three into a view helper ...

File: features/individual_charity_pages/charity_page.feature [hunk 2]
```diff
@@ -61,6 +61,18 @@ Feature: Web page owned by each charity
     Then I should not see "Health" within "what_they_do"
     And I should not see "Education" within "what_they_do"
     And I should not see "Animal Welfare" within "what_they_do"
+    
+  Scenario: display categories as links
+    Then I should see "Health" < tagged > with "a"
+    And  I should see "Education" < tagged > with "a"
+    And  I should see "Voluntary" < tagged > with "a"
+    And  I should see "Finance" < tagged > with "a"
+
```
Comment (tansaku) [inline features/individual_charity_pages/charity_page.feature]: the above scenario is probably redundant if we have the one below, and I wonder if we could reword the below so it didn't rely on the link stucture, i.e. we could click through and expect to see the particular orgs that are in the relevant category ...

File: spec/spec_helper.rb [hunk 3]
```diff
@@ -69,13 +69,13 @@
...
 =end
-  config.profile_examples = 10
+  # config.profile_examples = 10
 
   # Run specs in random order to surface order dependencies. If you find an
   # order dependency and want to debug it, you can fix the order by providing
   # the seed, which is printed after each run.
   #     --seed 1234
-  config.order = :random
+  # config.order = :random
```
Comment (tansaku) [inline spec/spec_helper.rb]: hmm, did we need to take this out?

File: features/individual_charity_pages/charity_page.feature (hunk 2 above)
Comment (tansaku) [inline features/individual_charity_pages/charity_page.feature]: there's a chance to pull the whole scenario to a much more declarative level if we are interested in that ...

http://itsadeliverything.com/declarative-vs-imperative-gherkin-scenarios-for-cucumber

https://danashby.co.uk/2014/12/19/declarative-or-imperative-how-do-you-write-your-feature-file-scenarios/

File: spec/spec_helper.rb (hunk 3 above)
Comment (bryszard) [inline spec/spec_helper.rb]: No, it just disturbed me when I ran single spec files (I wanted to see the order).

File: features/individual_charity_pages/charity_page.feature (hunk 2 above)
Comment (bryszard) [inline features/individual_charity_pages/charity_page.feature]: Now it's quite short. Not sure if that's what you wanted.

File: app/views/organisations/_show.html.erb (hunk 1 above)
Comment (bryszard) [inline app/views/organisations/_show.html.erb]: I'm not sure what kind of helper you see fit here. I created something, though I have now problems with adding comas (`join(', ')` is not working).

**Comment by tansaku** [comment]
excellent work here @bryszard

File: features/individual_charity_pages/charity_page.feature [hunk 4]
```diff
@@ -62,6 +62,11 @@ Feature: Web page owned by each charity
     And I should not see "Education" within "what_they_do"
     And I should not see "Animal Welfare" within "what_they_do"
 
+  Scenario: categories link to search results of orgs with these categories
+    Given I click "Health"
+    Then I should be on the organisations search page
+    And I should see "Friendly"
```
Comment (tansaku) [inline features/individual_charity_pages/charity_page.feature]: an alternative last step would be

```gherkin
And I should see only charities associated with "Health"
```

```ruby
Then \And I should see only charities associated with "(.*)"\ do |category|
   cat_orgs = Category.find_by(name: category).organisations
   cat_orgs.each do |org|
      expect(page).to have_contents (org.name)
   end
  (Organisations.all - cat_orgs).each do |org|
      expect(page).not_to have_contents (org.name)
   end
end
```

**Pull request merged**